static/app/components/tours/components.tsx

```diff
@@ -0,0 +1,226 @@
+
+interface TourContextProviderProps<T extends TourEnumType> {
+  /**
+   * The children of the tour context provider.
+   * All children of this component will be blurred when the tour is active.
+   * Only the active tour element and overlay will be discernible.
+   */
+  children: React.ReactNode;
+  /**
+   * Whether the tour can be accessed by the user
+   */
+  isAvailable: TourState<T>['isAvailable'];
+  /**
+   * The ordered list of Step IDs
+   */
+  orderedStepIds: TourState<T>['orderedStepIds'];
+  /**
+   * The React context (from createContext) containing the provider for the tour.
+   * The value for this prop comes from useTourReducer, to avoid extra steps.
+   */
+  tourContext: React.Context<TourContextType<T>>;
+}
+
+export function TourContextProvider<T extends TourEnumType>({
+  children,
+  isAvailable,
+  tourContext: TourContext,
+  orderedStepIds,
+}: TourContextProviderProps<T>) {
+  const tourContext = useTourReducer<T>({isAvailable, orderedStepIds, currentStep: null});
+  const isTourActive = tourContext.currentStep !== null;
+  return (
+    <TourContext.Provider value={tourContext}>
+      <BlurContainer>
+        {children}
+        {isTourActive && <BlurWindow />}
+      </BlurContainer>
+    </TourContext.Provider>
+  );
+}
+
+export interface TourElementProps<T extends TourEnumType>
+  extends Partial<UseOverlayProps> {
+  /**
+   * The content being focused during the tour.
+   */
+  children: React.ReactNode;
+  /**
+   * The description of the tour step.
+   */
+  description: React.ReactNode;
+  /**
+   * The unique identifier of the tour step.
+   */
+  id: TourStep<T>['id'];
+  /**
+   * The title of the tour step.
+   */
+  title: React.ReactNode;
+  /**
+   * The relevant tour context
+   */
+  tourContext: TourContextType<T>;
+}
+
+export function TourElement<T extends TourEnumType>({
+  children,
+  id,
+  title,
+  description,
+  tourContext,
+  position,
+}: TourElementProps<T>) {
+  const theme = useTheme();
+
+  const {dispatch, currentStep, orderedStepIds} = tourContext;
+  const stepCount = currentStep ? orderedStepIds.indexOf(id) + 1 : 0;
+  const stepTotal = orderedStepIds.length;
+  const hasPreviousStep = stepCount > 1;
+  const hasNextStep = stepCount < stepTotal;
+  const isOpen = currentStep?.id === id;
+
+  const {triggerProps, triggerRef, overlayProps} = useOverlay({isOpen, position});
+  const {current: element} = triggerRef;
+
+  useEffect(() => {
+    dispatch({
+      type: 'REGISTER_STEP',
+      step: {id, element},
+    });
+  }, [id, element, dispatch]);
+
+  const focusStyles = css`
+    position: relative;
+    z-index: ${theme.zIndex.toast};
+    &:after {
+      content: '';
+      position: absolute;
+      inset: 0;
+      border-radius: ${theme.borderRadius};
+      box-shadow: inset 0 0 0 3px ${theme.subText};
+    }
+  `;
+
+  return (
+    <Fragment>
+      <div css={isOpen ? focusStyles : undefined} {...triggerProps}>
```

Comment:
This `div` might cause issues if you need to render the element as an `li` or need it to have some specific positioning CSS. At the very least I'd allow `className`, but you could also do what `useHoverOverlay` does and allow `skipWrapper`